### PR TITLE
Support callable contextmanagers in contextlib

### PR DESCRIPTION
**Pull request by nimin98**
This commit is to support callable contextmanagers in contextlib, a.k.a., using contextmanager as a decorator. This pattern has been supported since Python 3.2 (https://github.com/python/cpython/blob/3.2/Lib/contextlib.py). As shown in the following code snippet, where `foo_cm` is a context manager. And it is used as a decorator to decorate function `foo`.

```
@contextmanager
def foo_cm(arg1):
    try:
        print(arg1)
        print('1')
        yield
    finally:
        print('2')

@foo_cm('0')
def foo():
    print('3')

foo()
```

The error we got from running mypy on the above code is as follows, 
```
error: ContextManager[Any] not callable
```

### CODE REVIEW

File: stdlib/2and3/contextlib.pyi [hunk 1]
```diff
@@ -20,7 +20,12 @@ class ContextManager(Generic[_T]):
                  exc_val: Optional[Exception],
                  exc_tb: Optional[TracebackType]) -> bool: ...
 
-def contextmanager(func: Callable[..., Iterator[_T]]) -> Callable[..., ContextManager[_T]]: ...
+if sys.version_info >= (3, 2):
+    class GeneratorContextManager(Generic[_T], ContextManager[_T]):
+        def __call__(self, Callable[..., _T]) -> Callable[..., _T]
```
Comment (JelleZijlstra) [inline stdlib/2and3/contextlib.pyi]: missing `: ...` (this is also breaking Travis)

File: stdlib/2and3/contextlib.pyi [hunk 2]
```diff
@@ -20,7 +20,12 @@ class ContextManager(Generic[_T]):
                  exc_val: Optional[Exception],
                  exc_tb: Optional[TracebackType]) -> bool: ...
 
-def contextmanager(func: Callable[..., Iterator[_T]]) -> Callable[..., ContextManager[_T]]: ...
+if sys.version_info >= (3, 2):
+    class GeneratorContextManager(Generic[_T], ContextManager[_T]):
```
Comment (JelleZijlstra) [inline stdlib/2and3/contextlib.pyi]: You don't need to explicitly inherit from Generic.

Comment (nimin98) [inline stdlib/2and3/contextlib.pyi]: I was following the pattern used in this same file, for example on line 28,

```
class closing(Generic[_T], ContextManager[_T]):
    def __init__(self, thing: _T) -> None: ...
```

**Comment by ambv** [comment]
Thanks! ✨ 🍰 ✨

**Pull request merged**